common/reconciliation/invariants/concreteExecutionExists.go

```diff
@@ -59,6 +60,11 @@ func (c *concreteExecutionExists) Check(execution interface{}) common.CheckResul
 			InvariantType:   c.InvariantType(),
 		}
 	}
+	var runIDCheckResult *common.CheckResult
+	currentExecution, runIDCheckResult = c.validateCurrentRunID(currentExecution)
+	if runIDCheckResult != nil {
+		return *runIDCheckResult
+	}
 	// by this point the corresponding concrete execution must exist and can be already closed
 	currentExecResp, currentExecErr := c.pr.IsWorkflowExecutionExists(&persistence.IsWorkflowExecutionExistsRequest{
```

Comment:
Also not related to your change, but I think there's a race condition here. The workflow retention period can kick in before we make this call, so even if the response says the concrete execution doesn't exist, we can't be 100% sure it's a corruption, we need to check if the current execution with the same runID is still there.

another nit: can we rename the var from `currentExecResp, currentExecErr` to `concreteExceResp, concreteExecErr`?

Reply:
Yes. You are right. The deletion of the current execution will check the current run ID.

I will update the namings.

Reply:
the retention kicks in after the workflow gets closed only but there is a check for the open state above.

Reply:
Synced offline. We will also cover the case where state is closed.